jax/_src/cudnn/fused_attention_stablehlo.py

```diff
@@ -978,14 +982,11 @@ def dot_product_attention(query: Array,
     bias = bias.reshape((1,) * (4 - len(bias.shape)) + bias.shape)
   # check if input shape and data type is compatiable
   check_layout(query, key, value, bias, q_seqlen, kv_seqlen, layout)
-  # check if flash attention is supported for this attention pattern
-  check_is_flash_attention(
-      query, key, layout, cudnn_version, bias is not None, is_training)
   if has_padding(mask_type) and (q_seqlen is None or kv_seqlen is None):
     raise ValueError("Require q_seqlen and kv_seqlen to generate padding mask")
   has_bias = bias is not None
   has_mask = mask is not None
-  has_dbias = has_bias and is_training and \
+  has_dbias = has_bias and \
```

Comment:
The remove of `is_training` from here changes the behavior, right?

Also, I feel the better way to do this is not to pass the `variadic_args` around. Instead, we can compute it in `_dot_product_attention_fwd` and `_dot_product_attention_fwd_rule` respectively, where we should be able to access the `bias`, `mask`, `query` and `layout`, right?

Reply:
It does not change the behavior, the has_dbias is only used in bwd anyway and is_training is always True. Also bias/mask is set to one if it is None as Jax does not accept None here so it is better to compute before.